src/bridge/ISequencerInbox.sol

```diff
@@ -220,10 +229,33 @@ interface ISequencerInbox is IDelayedMessageProvider {
      */
     function setBatchPosterManager(address newBatchPosterManager) external;
 
+    /**
+     * @notice Updates the fee token pricer, the contract which is used to get the exchange rate between child
+     *         chain's fee token and parent chain's native token in rollups that use custom fee token.
+     * @param newFeeTokenPricer The new fee token pricer to be set
+     */
+    function setFeeTokenPricer(IFeeTokenPricer newFeeTokenPricer) external;
+
     /// @notice Allows the rollup owner to sync the rollup address
     function updateRollupAddress() external;
 
     // ---------- initializer ----------
 
-    function initialize(IBridge bridge_, MaxTimeVariation calldata maxTimeVariation_) external;
+    function initialize(
+        IBridge bridge_,
+        MaxTimeVariation calldata maxTimeVariation_,
+        IFeeTokenPricer feeTokenPricer_
+    ) external;
+}
+
+// CHRIS: TODO: analyse possible effects of a malicious fee token pricer
+// CHRIS: TODO: should the interface be a view function?
+interface IFeeTokenPricer {
+    /**
+     * @notice Get the number of child chain's fee tokens per 1 parent chain's native token. Exchange rate must be
+     *         denominated in 18 decimals.
+     * @dev    For example, parent chain's native token is ETH, fee token is DAI. If price of 1ETH = 2000DAI, then function should return 2000*1e18.
+     *         If fee token is USDC instead and price of 1ETH = 2000USDC, function should still return 2000*1e18, no matter that USDC uses 6 decimals.
+     */
+    function getExchangeRate() external returns (uint256);
```

Comment:
instead of a getter that return spot rate, I think it make more sense to have a mutable method `spendingInEth(uint256 wei, address spender)` that take the number of eth spent as input and return the number of token spent. This allow the pricer to do more complex internal accounting if needed.